src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs

```diff
@@ -417,6 +416,30 @@ private bool ShouldBeInFullSyncMode(Snapshot best)
             return result;
         }
 
+        private bool ShouldBeInFullSyncModeInArchiveMode(Snapshot best)
+        {
+            bool notInBeaconModes = !best.IsInAnyBeaconMode;
+            bool desiredPeerKnown = AnyDesiredPeerKnown(best);
+            bool notInStateSync = !best.IsInStateSync;
```

Comment:
notInStateSync & notNeedToWaitForHeaders what is the sense of this conditions in archive mode?

Reply:
Don't know. I just remove any fast sync .

Reply:
I don't know. I just remove any use of sync pivot.

Reply:
I think we should remove it (cc: @LukaszRozmej )

Reply:
Yes we can drop it here earlier I was reusing method that was also used in fast sync so it was needed there.

Reply:
Pushed that